### PR TITLE
FIX: supplier invoice: import only selected origin lines

### PR DESCRIPTION
**Pull request by marc-dll**

### CODE REVIEW

**Comment by marc-dll** [comment]
Open to discussion about this PR, but you have to say something first ;)

File: htdocs/fourn/facture/card.php [hunk 1]
```diff
@@ -978,6 +979,10 @@
 							$num = count($lines);
 							for ($i = 0; $i < $num; $i++) // TODO handle subprice < 0
 							{
+                                if (! in_array($lines[$i]->id, $selectedLines)) {
```
[the remark this answers is not in the document]
Comment (eldy) [reply htdocs/fourn/facture/card.php:982]: Can you confirm me that all lines are checked by default ? If not, we should exclude non selected only if there is at least one selection. If yes, fix can stay as it is.

[the remark this answers is not in the document]
Comment (moreauftheobald) [reply htdocs/fourn/facture/card.php:982]: Hello, The pull request has been done for my company, i've just checked, and i can grant that all the checkbox are checked by default on the invoice cretation screen.

[the remark this answers is not in the document]
Comment (marc-dll) [reply htdocs/fourn/facture/card.php:982]: @eldy I concurr, they are all selected by default at first.

**Comment by marc-dll** [comment]
Closing because #18860 has been merged for the same subject ;)

**Pull request closed without merging**